include/bluetooth/gatt.h

```diff
@@ -1669,6 +1669,13 @@ struct bt_gatt_subscribe_params {
 #endif /* CONFIG_BT_GATT_AUTO_DISCOVER_CCC */
 	/** Subscribe value */
 	uint16_t value;
+#if defined(CONFIG_BT_SMP)
+	/** Minimum required security for received notification. If not set
+	 * explicitly by application it is set by host based on security level
+	 * at the time of subscription.
+	 */
+	bt_security_t min_security;
```

Comment:
I think the implementation matches the requirement from the specification, but it only seems like a odd requirement; why should the client be responsible for discarding notifications with lower security? E.g. if a notification is sent unencrypted when it should have been encrypted, having the client discard it makes no sense, and won't solve any issues. If anything, the client should increase security when that happens.

Reply:
I think this is precaution for bogus devices, until encryption is enabled you cannot know if peer is valid, properly behaving servers won't send notifications if sec level is too low

Reply:
Ah, yes, that makes sense :)